nemo_curator/utils/distributed_utils.py

```diff
@@ -415,15 +421,30 @@ def read_data(
             df = df.to_backend("cudf")
 
     elif file_type in ["json", "jsonl", "parquet"]:
+        assert len(input_files) > 0
+
+        input_extensions = {os.path.splitext(f)[-1] for f in input_files}
+        if len(input_extensions) != 1:
+            raise RuntimeError(
```

Comment:
An example of when we would expect this `RuntimeError` is for:
```
doc = DocumentDataset.read_json(in_files)
```
Where `in_files` is a string path to a directory with multiple JSONL files and a CRC file. Since the CRC file is not explicitly being filtered out, we raise the error.

Reply:
We can leave this as is for now. 
In theory there might be cases where a user filters by `[.json, .jsonl]` using the file filter, but will raise errors here. In practice I expect it to be unlikely so we can wait an see if there is any user feedback around this.